### PR TITLE
Fix chef-server-ctl backup always returning 1

### PR DESCRIPTION
**Pull request by ryancragun**
Fixes https://github.com/chef/chef-server/issues/534
http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/721/downstreambuildview/

### CODE REVIEW

**Comment by stevendanna** [comment]
I'm :+1: on just merging; however, if you have a few minutes, here are some things I had questions about:

1) Error output should probably go to standard error rather than standard out.
2) Given our purpose, I think the bare rescue makes sense, but I'd prefer to see it just around foreign code (ChefBackup::Runner).  The other code we are wrapping looks to be related to transforming user-input into configuration.  If something is wrong with those cases, I think we are going to want something more descriptive than the ruby exception message since it probably means the user has input they need to fix.

File: omnibus/files/private-chef-ctl-commands/backup.rb [hunk 1]
```diff
@@ -25,10 +25,15 @@
...
+    Chef::Mixin::DeepMerge.deep_merge!(stringify_keys(options.to_h), running_config['private_chef']['backup'])
+    ChefBackup::Runner.new(running_config).backup
+  rescue => e
+    puts "[ERROR] #{e.message}"
+    exit(1)
+  end
 
-  status = ChefBackup::Runner.new(running_config).backup
-  exit(status ? 0 : 1)
+  exit(0)
```
Comment (marcparadise) [inline omnibus/files/private-chef-ctl-commands/backup.rb]: the way omnibus-ctl works you can just 'return 0' , and above 'return 1'. That will be used as the exit code.

**Comment by marcparadise** [comment]
@stevendanna to the best of my recollection, all of the omnibus-ctl error output (except wrapped chef runs) go to stdout.

**Comment by ryancragun** [comment]
@stevendanna All error output now goes through STDERR and we only rescue on the foreign code.
@marcparadise I've seen early returns in omnibus-ctl cause jump exceptions because omnibus-ctl will try and yield a non-existent block, hence the specific exits.

**Comment by stevendanna** [comment]
Thanks for the updates, :+1: from me.  Will merge at EOD if there are no objections

**Pull request merged**